### PR TITLE
Replace RestAssured.when with Spring WebTestClient

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #221

### CODE REVIEW

**Comment by geoand** [comment]
@spencergibb Is this what you had in mind?

**Comment by geoand** [comment]
Should I update the PR or should this be done later?

File: pom.xml [hunk 1]
```diff
@@ -123,10 +123,9 @@
 			</dependency>
 
 			<dependency>
-				<groupId>io.rest-assured</groupId>
-				<artifactId>rest-assured</artifactId>
-				<version>${restassured.version}</version>
-				<scope>test</scope>
+				<groupId>org.projectreactor</groupId>
+				<artifactId>reactor-spring</artifactId>
```
Comment (spencergibb) [inline pom.xml]: Where is this used? All the imports seem to be boot or framework specific.

Comment (geoand) [reply pom.xml]: I was getting a weird exception from the client when this dependency wasn't present

**Comment by ryanjbaxter** [comment]
I would take care of the imports now before it is merged

**Comment by geoand** [comment]
@ryanjbaxter Fixed :)

**Comment by spencergibb** [comment]
Closed with some polish via 8be84431bc2e2a64122846e27a206cff3e99bd17

**Comment by spencergibb** [comment]
I was able to remove the `reactor-spring` dependency without a problem.

**Comment by geoand** [comment]
@spencergibb great to hear! Thanks!

**Pull request closed without merging**